app/models/ep_postmaster/mailgun_post.rb

```diff
@@ -2,24 +2,31 @@
 
 module EpPostmaster
   class MailgunPost
-    attr_accessor :message_id, :x_mailgun_sid, :code, :message_headers, :domain, :error, :event, :recipient, :reply_to, :subject, :signature, :timestamp, :token, :from, :sender
+
+    attr_accessor :message_id, :code, :message_headers, :domain, :error, :event, :recipient, :reply_to, :subject, :signature, :timestamp, :token, :from, :sender, :reason
 
     def initialize(params)
-      @message_id = params["message-id"]
-      @x_mailgun_sid = params["X-Mailgun-Sid"]
-      @code = params["code"]
-      @message_headers = JSON.parse(params["message-headers"]) rescue {}
-      @domain = params["domain"]
-      @error = params["error"]
-      @event = params["event"]
-      @recipient = params.fetch("recipient")
-      @reply_to = find_reply_to
-      @subject = find_subject
-      @from = Array(message_headers.select { |header| header[0] == "From" }.first)[1]
-      @sender = Array(message_headers.select { |header| header[0] == "Sender" }.first)[1]
-      @signature = params["signature"]
-      @timestamp = params["timestamp"]
-      @token = params["token"]
+      event_data = params["event-data"] || {}
+      @message_headers = event_data["message"]["headers"] || {}
```

Comment:
If `event_data` can be `{}` here, might be worth a `dig` to get the headers so we don't get a `no method [] for nilClass` looking for `"headers"`.
```ruby
@message_headers = event_data.dig("message", "headers") || {}
```